src/Umbraco.Tests/Testing/UmbracoTestBase.cs

```diff
@@ -418,15 +421,25 @@ protected virtual void Reset()
             // reset and dispose scopes
             // ensures we don't leak an opened database connection
             // which would lock eg SqlCe .sdf files
-            if (Factory?.TryGetInstance<IScopeProvider>() is ScopeProvider scopeProvider)
+
+            try
             {
-                Scope scope;
-                while ((scope = scopeProvider.AmbientScope) != null)
+                var instance = Factory?.TryGetInstance<IScopeProvider>();
+                if (instance is ScopeProvider scopeProvider)
                 {
-                    scope.Reset();
-                    scope.Dispose();
+                    Scope scope;
+                    while ((scope = scopeProvider.AmbientScope) != null)
+                    {
+                        scope.Reset();
+                        scope.Dispose();
+                    }
                 }
             }
+            catch (ResolveUnregisteredDependencyException)
```

Comment:
Is this something we'd have to worry about during regular application shutdown?

Reply:
I don't believe so no. I triggered a shutdown with a web.config change whilst debugger attached and I can't get the fallback to throw. Logs look good also.

Essentially it just occurs in the tests now because previously line 427 in UmbracoTestBase would get a null and carry on whilst skipping scope.Reset() && Dispose in tests that handn't wired up a mock scope provider

```csharp
var instance = Factory?.TryGetInstance<IScopeProvider>();
```

A good demo of the issue occuring is in

Umbraco.Tests.Models.ContentTests #Can_Change_Property_Value 

TearDown in UmbracoTestBase will trigger the exception with "Unable to resolve type: Umbraco.Core.Scoping.IScopeProvider"